keras_cv/layers/preprocessing/random_flip.py

```diff
@@ -0,0 +1,178 @@
+    `mode` attribute. During inference time, the output will be identical to
+    input. Call the layer with `training=True` to flip the input.
+
+    Input pixel values can be of any range (e.g. `[0., 1.)` or `[0, 255]`) and
+    of integer or floating point dtype. By default, the layer will output
+    floats.
+
+    Input shape:
+      3D (unbatched) or 4D (batched) tensor with shape:
+      `(..., height, width, channels)`, in `"channels_last"` format.
+
+    Output shape:
+      3D (unbatched) or 4D (batched) tensor with shape:
+      `(..., height, width, channels)`, in `"channels_last"` format.
+
+    Arguments:
+      mode: String indicating which flip mode to use. Can be `"horizontal"`,
+        `"vertical"`, or `"horizontal_and_vertical"`. Defaults to
+        `"horizontal_and_vertical"`. `"horizontal"` is a left-right flip and
+        `"vertical"` is a top-bottom flip.
+      seed: Integer. Used to create a random seed.
+    """
+
+    def __init__(
+        self,
+        mode=HORIZONTAL_AND_VERTICAL,
+        seed=None,
+        bounding_box_format=None,
+        **kwargs
+    ):
+        super().__init__(seed=seed, force_generator=True, **kwargs)
+        self.mode = mode
+        if mode == HORIZONTAL:
+            self.horizontal = True
+            self.vertical = False
+        elif mode == VERTICAL:
+            self.horizontal = False
+            self.vertical = True
+        elif mode == HORIZONTAL_AND_VERTICAL:
+            self.horizontal = True
+            self.vertical = True
+        else:
+            raise ValueError(
+                "RandomFlip layer {name} received an unknown mode="
+                "{arg}".format(name=self.name, arg=mode)
+            )
+        self.auto_vectorize = False
+        self.bounding_box_format = bounding_box_format
+
+    def augment_label(self, label, transformation, **kwargs):
+        return label
+
+    def augment_image(self, image, transformation, **kwargs):
+        image = utils.preprocessing.ensure_tensor(image, self.compute_dtype)
+        flipped_output = image
+        if self.horizontal and transformation["flip_horizontal"]:
+            flipped_output = tf.image.flip_left_right(flipped_output)
+        if self.vertical and transformation["flip_vertical"]:
+            flipped_output = tf.image.flip_up_down(flipped_output)
+        flipped_output.set_shape(image.shape)
+        return flipped_output
+
+    def get_random_transformation(self, **kwargs):
+        flip_horizontal = False
+        flip_vertical = False
+        if self.horizontal:
+            flip_horizontal = np.random.choice([True, False])
+        if self.vertical:
+            flip_vertical = np.random.choice([True, False])
+        return {
+            "flip_horizontal": flip_horizontal,
+            "flip_vertical": flip_vertical,
+        }
+
+    def augment_bounding_boxes(self, bounding_boxes, transformation=None, **kwargs):
+        image = None
```

Comment:
you can replace `image = None` and `image = kwargs.get` with a keyward argument in augment_bounding_boxes().